influxql/functions.gen.go

```diff
@@ -377,6 +380,50 @@ func (r *FloatElapsedReducer) Emit() []IntegerPoint {
 	return nil
 }
 
+// FloatSampleReduces implements a reservoir sampling to calculate a random subset of points
+type FloatSampleReducer struct {
+	count int // how many points we've iterated over
+
+	points floatPoints // the reservoir
+}
+
+// NewFloatSampleReducer creates a new FloatSampleReducer
+func NewFloatSampleReducer(size int) *FloatSampleReducer {
+	return &FloatSampleReducer{
+		points: make(floatPoints, size),
+	}
+}
+
+// AggregateFloat aggregates a point into the reducer.
+func (r *FloatSampleReducer) AggregateFloat(p *FloatPoint) {
+	// Fill the reservoir with the first n points
+	if r.count < len(r.points) {
+		r.points[r.count] = *p
+		r.count++
```

Comment:
Regardless, I'll change it so that I increment the counter at the start

```
	if r.count < len(r.points) {
		r.points[r.count-1] = *p
		r.count++
		return
	}
```